src/array.jl

```diff
@@ -193,18 +193,21 @@ beamform(s, f, sd; fs=framerate(s), method=Bartlett()) = beamform(s, f, 1, sd; f
 
 function beamformer(::Bartlett, R, sv)
   # TODO: add support for shading
-  [abs.(sv[:,j]' * R * sv[:,j]) for j ∈ 1:size(sv,2)]
+  RS = R * sv
+  pseudospectrum = zeros(Float64, size(sv, 2))
+  pseudospectrum .= abs.(dot.(eachcol(sv), eachcol(RS)))
 end
 
 function beamformer(opt::Capon, R, sv)
   opt.γ != 0 && (R .+= opt.γ * I(size(R, 1)))
-  [1 ./ abs.(sv[:,j]' * inv(R) * sv[:,j]) for j ∈ 1:size(sv,2)]
+  pseudospectrum = zeros(Float64, size(sv, 2))
+  RiS = inv(Hermitian(R)) * sv
+  pseudospectrum .= 1 ./ abs.(dot.(eachcol(sv), eachcol(RiS)))
 end
 
 function beamformer(opt::Music, R, sv)
-  F = eigen(R)
-  ndx = sortperm(F.values; rev=true)
-  Q = F.vectors[:, ndx[opt.nsignals+1:end]]
-  Q = Q * Q'
-  [1 ./ abs.(sv[:,j]' * Q * sv[:,j]) for j ∈ 1:size(sv,2)]
+  pseudospectrum = zeros(Float64, size(sv, 2))
+  _, Q = eigen(Hermitian(R), 1:(size(R, 1)-opt.nsignals))
+  QS = Q' * sv
+  pseudospectrum .= 1 ./ real.(dot.(eachcol(QS), eachcol(QS)))
```

Comment:
```suggestion
  _, Q = eigen(Hermitian(R), 1:(size(R, 1)-opt.nsignals))
  QS = Q' * sv
  1 ./ real.(dot.(eachcol(QS), eachcol(QS)))
```

Verification:
```julia
julia> F = eigen(R); ndx = sortperm(F.values; rev=true); Q = F.vectors[:, ndx[1+1:end]]; Q = Q * Q'; x = [1 ./ abs.(sv[:,j]' * Q * sv[:,j]) for j ∈ 1:size(sv,2)];

julia> _, Q = eigen(Hermitian(R), 1:(size(R, 1)-1)); QS = Q' * sv; y = 1 ./ real.(dot.(eachcol(QS), eachcol(QS)));

julia> x ≈ y
true
```

Reply:
By the suggested change I think you'll end up with an extra allocation. I think where we will see problems is for larger arrays where the size and the types of the array are unknown.  Probably need to add types to the signatures at some point too

Reply:
I don't think there is any extra allocation.

Benchmark:
```julia
julia> function f1(R, sv)
         pseudospectrum = zeros(Float64, size(sv, 2))
         _, Q = eigen(Hermitian(R), 1:(size(R, 1)-1))
         QS = Q' * sv
         pseudospectrum .= 1 ./ real.(dot.(eachcol(QS), eachcol(QS)))
       end
f1 (generic function with 1 method)

julia> function f2(R, sv)
         _, Q = eigen(Hermitian(R), 1:(size(R, 1)-1))
         QS = Q' * sv
         1 ./ real.(dot.(eachcol(QS), eachcol(QS)))
       end
f2 (generic function with 1 method)

julia> sv = randn(64, 256);

julia> R = let A = randn(64,64)
         A + A'
       end;

julia> @btime f1(R, sv);
  2.009 ms (28 allocations: 247.75 KiB)

julia> @btime f2(R, sv);
  2.008 ms (28 allocations: 247.75 KiB)

julia> f1(R, sv) == f2(R, sv)
true
```

Reply:
To benchmark the function I think you might need to call the beamformer and use a larger sensor array with more samples. Benchmarking the inner calls individually with small sizes will yield similar results. 

I'll probably setup some benchmarking at some point so it's easier to compare between branches using the same testing data. It's sitting on my machine I just need to clean it up

Reply:
For now, we will go with what we have on the branch now. When you have an update, do raise a PR if needed.